components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/listener/TenantCreationEventListener.java

```diff
@@ -91,8 +91,11 @@ public int getListenerOrder() {
 
     @Override
     public void onPreDelete(int tenantId) throws StratosException {
-
-        revokeTokens(tenantId);
+        try {
+            OAuth2Util.deleteAllAccessTokens(tenantId);
```

Comment:
Why did we decide to introduce a new method other than using the revokeTokens(tenantId) method?

Reply:
`revokeTokens` only updates the state of the tokens. The new method was introduced to delete/clean all the tokens of a particular tenant in DB level.

Reply:
I think it is fine to just call revokeTokens... There is a task to delete unused tokens. That will take care deleting records.

Reply:
+1
Also, I think when we change the state to revoked tokens are automatically deleted and move to an audit table (not sure whether this feature is enabled by default in IS)

Reply:
Also at this layer, we are not invoking any listeners nor triggering a revoke event.
I think it would be better to do so. 

There are other places that are missing this too. Like when we delete an SP etc.